site/cds_rdm/inspire_harvester/transform_entry.py

```diff
@@ -0,0 +1,248 @@
+        return files
+
+    def _parent(self):
+        """Record parent minimal values."""
+        return {
+            "access": {
+                "owned_by": {
+                    "user": 2,  # temporary solution before we have System user
+                }
+            }
+        }
+
+    def _access(self):
+        """Record access minimal values."""
+        return {
+            "record": "public",
+            "files": "public",
+        }
+
+    def build(self):
+        """BPerform building of CDS-RDM entry record."""
+        inspire2cds = Inspire2RDM(self.inspire_metadata)
+        rdm_record = {
+            "metadata": self._metadata(inspire2cds),
+            "files": self._files(inspire2cds),
+            "parent": self._parent(),
+            "access": self._access(),
+        }
+
+        return rdm_record, self.errors
+
+
+class Inspire2RDM:
+    """INSPIRE to CDS-RDM record mapping."""
+
+    def __init__(self, inspire_metadata):
+        """Initializes the Inspire2RDM class."""
+        self.inspire_metadata = inspire_metadata
+        self.metadata_errors = []
+        self.files_errors = []
+
+    def _transform_titles(self):
+        """Mapping of INSPIRE titles to metadata.title and additional_titles."""
+        inspire_titles = self.inspire_metadata.get("titles")
+        rdm_title = None
+        rdm_additional_titles = []
+
+        for i, inspire_title in enumerate(inspire_titles):
+            try:
+                if i == 0:
+                    rdm_title = inspire_title.get("title")
+                else:
+                    rdm_additional_titles.append(
+                        {
+                            "title": inspire_title.get("title"),
+                            "type": {
+                                "id": "alternative-title",
+                            },
+                        }
+                    )
+
+                if inspire_title.get("subtitle"):
+                    rdm_additional_titles.append(
+                        {
+                            "title": inspire_title.get("subtitle"),
+                            "type": {
+                                "id": "subtitle",
+                            },
+                        }
+                    )
+            except Exception as e:
+                self.metadata_errors.append(
+                    f"Error occurred while mapping titles. Title from INSPIRE: {inspire_title}. INSPIRE record id: {self.inspire_metadata.get('control_number')}. Error: {e}."
+                )
+                return None, None
+
+        return rdm_title, rdm_additional_titles
+
+    def _transform_publication_date(self):
+        """Mapping of INSPIRE thesis_info.date to metadata.publication_date."""
+        thesis_date = self.inspire_metadata.get("thesis_info", {}).get(
+            "date"
+        ) or self.inspire_metadata.get("thesis_info", {}).get("defense_date")
+        try:
+            parsed_date = str(parse_edtf(thesis_date))
```

Comment:
why double conversion (edtf and then str) is needed?

Reply:
It was needed because `parse_edtf` returns `class 'edtf.parser.parser_classes.Date'` and I needed a string